src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -64,7 +64,16 @@ public IStringLocalizer Create(Type resourceSource)
 
             var typeInfo = resourceSource.GetTypeInfo();
             var assembly = typeInfo.Assembly;
-            var baseName = _applicationEnvironment.ApplicationName + "." + _resourcesRelativePath + resourceSource.Name;
+
+            string baseName = string.Empty;
+            if (!string.IsNullOrEmpty(_resourcesRelativePath))
```

Comment:
I would do neither. String interpolation adds no value here, and the `if()` is indeed not necessary. I'd just do `baseName = _applicationEnvironment.ApplicationName + "." + _resourcesRelativePath + typeInfo.FullName;` always.